src/components/Swap/Toolbar/index.tsx

```diff
@@ -79,6 +79,18 @@ export default memo(function Toolbar() {
     return inputBalance && inputAmount && inputBalance.lessThan(inputAmount)
   }, [inputAmount, inputBalance])
 
+  const maybeToggleOpen = useCallback(() => {
+    if (
+      !insufficientBalance &&
+      !isWrap &&
+      !(state === TradeState.NO_ROUTE_FOUND || (trade && !trade.swaps)) &&
+      trade?.inputAmount &&
+      trade?.outputAmount
+    ) {
+      onToggleOpen()
+    }
+  }, [insufficientBalance, isWrap, onToggleOpen, state, trade])
+
   const caption = useMemo(() => {
```

Comment:
Instead of duplicating the logic, just return a pair, `[caption, isTogglable]`

Reply:
I returned an object instead of a pair because arrays of ReactNodes require keys.